src/commonMain/kotlin/fr/acinq/lightning/channel/Commitments.kt

```diff
@@ -270,6 +270,11 @@ data class Commitments(
             return Either.Left(TooManyAcceptedHtlcs(channelId, maximum = commitments1.remoteParams.maxAcceptedHtlcs.toLong()))
         }
 
+        // README: this is not part of the LN Bolts: we also check against our own limit, to avoid creating commit txs that have too many outputs
+        if (outgoingHtlcs.size > commitments1.localParams.maxAcceptedHtlcs) {
```

Comment:
That means the maximum number of HTLC we'll have in the commit tx is `2 * localParams.maxAcceptedHtlcs` (since we have at most `localParams.maxAcceptedHtlcs` incoming HTLCs + at most `localParams.maxAcceptedHtlcs` outgoing HTLCs), is that ok?

Reply:
Yes. But maybe it would be cleaner to add a specific `max-offered-htlcs` parameter to`NodeParams`, I'll check how it impacts serialisation and peer backups.

Reply:
To be honest too many configuration hooks can become overkill...
I would rather make it clear that the total number of HTLCs in the commit tx can be twice `max-accepted-htlcs` and use just that for now.

Reply:
Ok I've changed my mind and I think using `max-accepted-htlcs` for both received and offered payments makes more sense :)

Reply:
I think we also need to apply `max-htlc-value-in-flight` symmetrically, don't you think?

Reply:
Yes it makes sense